casper/src/main/scala/coop/rchain/casper/BlockValidationLogic.scala

```diff
@@ -113,6 +113,28 @@ object BlockValidationLogic {
       .getOrElse(BlockStatus.valid.asRight[InvalidBlock])
   }
 
+  def transactionExpiration(
+      b: BlockMessage,
+      expirationThreshold: Int,
+      expiredDeployF: String => Unit
+  ): ValidBlockProcessing = {
+    val earliestAcceptableValidAfterBlockNumber = b.blockNumber - expirationThreshold
+    val deploys                                 = b.state.deploys.map(_.deploy)
+    val maybeExpiredDeploy =
+      deploys.find(_.data.validAfterBlockNumber <= earliestAcceptableValidAfterBlockNumber)
+    maybeExpiredDeploy
```

Comment:
In many places in the code Haskell naming convention where Option is called Maybe is used.
But since this is Scala code and its Option we are using for quite a while `Opt` suffix to define optional values. So it should be expiredDeployOpt.